### PR TITLE
feat(UserAgent): add user agent string

### PR DESCRIPTION
**Pull request by tkrugg**
⚠️ ~awaiting support for X-Algolia-Agent URL param https://github.com/algolia/go/pull/1189~ it's been merged

### CODE REVIEW

**Comment by Haroenv** [comment]
Looks good to me if the URL is accepted

File: lib/_sendEvent.ts [hunk 1]
```diff
@@ -16,6 +17,9 @@ export type InsightsEvent = {
   filters?: string[];
 };
 
+const USER_AGENT = encodeURIComponent(
+  `Algolia insights for vanilla JavaScript (${version})`
```
Comment (samouss) [inline lib/_sendEvent.ts]:
```suggestion
  `Algolia insights for JavaScript (${version})`
```

I don't think the `vanilla` is important. I've removed it from the API Client.

Comment (tkrugg) [reply lib/_sendEvent.ts]: done here, thanks: 1ef66dc

File: lib/__tests__/_sendEvent.test.ts [hunk 2]
```diff
@@ -54,6 +55,20 @@ describe("sendEvent", () => {
         ]
       });
     });
+    it("should include X-Algolia-* query parameters", () => {
+      (AlgoliaInsights as any).sendEvent("click", {
+        eventName: "my-event",
+        index: "my-index",
+        objectIDs: ["1"]
+      });
+      const [, requestUrl] = XMLHttpRequest.open.mock.calls[0];
```
Comment (Haroenv) [inline lib/__tests__/_sendEvent.test.ts]:
```suggestion
      const requestUrl = XMLHttpRequest.open.mock.calls[0][1];
```

I don't see much advantage of the destructuring here TBH, but no big deal either way

Comment (tkrugg) [reply lib/__tests__/_sendEvent.test.ts]: remove there and in another place: 3cf7fd2d7da925b2b04da02acc324e80566360f0

**Comment by tkrugg** [comment]
@samouss yup, kept this very simple for now

**Pull request merged**